src/View/Helper/FormHelper.php

```diff
@@ -657,21 +662,34 @@ protected function _containerOptions(?string $fieldName, array $options): array
         ) {
             $options['container'] = $this->injectClasses('form-floating', (array)($options['container'] ?? []));
         }
+        $skipFormGroupClass = ($options['container']['formGroupClass'] ?? null) === false;
 
-        if (!isset($options['container'])) {
-            return $options;
+        if ($skipFormGroupClass) {
+            unset($options['container']['formGroupClass']);
+            if (empty($options['container'])) {
+                unset($options['container']);
+            }
         }
 
         $containerOptions = $options['container'];
```

Comment:
This line will generate an `undefined index` error if the `unset()` on line 670 is executed. Test case for when `$skipFormGroupClass` ends up being `true` is needed.

P.S. I have fixed the test suite issues on upstream master, so you can rebase.